st2api/st2api/controllers/exp/inquiries.py

```diff
@@ -0,0 +1,295 @@
+
+    # No data model currently exists for Inquiries, so we're "borrowing" ActionExecutions
+    model = ActionExecutionAPI
+    access = ActionExecution
+
+    def get_all(self, requester_user=None, limit=None, **raw_filters):
+        """Retrieve multiple Inquiries
+
+           TODO(mierdin): This function doesn't currently make RBAC assertions for two reasons:
+           - The equivalent for executions (Resource._get_all) doesn't seem to do this either.
+           - I am not sure the best approach, since I would have to make an assertion
+             inside the below loop?
+
+            Handles requests:
+                GET /inquiries/
+        """
+
+        raw_inquiries = super(InquiriesController, self)._get_all(
+            limit=limit,
+            raw_filters={'status': 'pending', 'runner': 'inquirer'}
+        )
+
+        inquiries = []
+        for raw_inquiry in json.loads(raw_inquiries.body):
+
+            # _get_all includes workflows that also contain executions
+            # with 'pending' status, so we want to prune all workflows
+            # (The action won't have a 'children' key)
+            if raw_inquiry.get('children'):
+                continue
+
+            inquiries.append(self._transform_inquiry(raw_inquiry))
+
+        return inquiries
+
+    def get_one(self, inquiry_id, requester_user=None):
+        """Retrieve a single Inquiry
+
+            Handles requests:
+                GET /inquiries/<inquiry id>
+        """
+
+        # Retrieve the desired inquiry
+        #
+        # (Passing permission_type here leverages _get_one_by_id's built-in
+        # RBAC assertions)
+        raw_inquiry = self._get_one_by_id(
+            id=inquiry_id,
+            requester_user=requester_user,
+            permission_type=PermissionType.INQUIRY_VIEW
+        )
+
+        if raw_inquiry.runner.get('runner_module') != "inquirer":
+            abort(http_client.BAD_REQUEST, '%s is not an Inquiry.' % inquiry_id)
+            return
+
+        if raw_inquiry.status != "pending":
+            abort(http_client.BAD_REQUEST, 'Inquiry %s has already been responded to' % inquiry_id)
+            return
+
+        new_inquiry = self._transform_inquiry(raw_inquiry)
+
+        return new_inquiry
+
+    def put(self, inquiry_id, response_data, requester_user):
+        """Provide response data to an Inquiry
+
+            In general, provided the response data validates against the provided
+            schema, and the user has the appropriate permissions to respond,
+            this will set the Inquiry execution to a successful status, and resume
+            the parent workflow.
+
+            Handles requests:
+                PUT /inquiries/<inquiry id>
+        """
+
+        LOG.debug("Inquiry %s received response payload: %s" % (inquiry_id, response_data.response))
+
+        # Retrieve details of the inquiry via ID (i.e. params like schema)
+        inquiry_execution = self._get_one_by_id(
```

Comment:
How does permission check work right now? Does user need to have a permission to a specific inquiry id to respond?

Reply:
Please ignore that - I see "_can_respond" method below.